homeassistant/components/mqtt/water_heater.py

```diff
@@ -0,0 +1,307 @@
+    CONF_CURRENT_TEMP_TOPIC,
+    CONF_MODE_COMMAND_TEMPLATE,
+    CONF_MODE_COMMAND_TOPIC,
+    CONF_MODE_LIST,
+    CONF_MODE_STATE_TEMPLATE,
+    CONF_MODE_STATE_TOPIC,
+    CONF_PRECISION,
+    CONF_RETAIN,
+    CONF_TEMP_COMMAND_TEMPLATE,
+    CONF_TEMP_COMMAND_TOPIC,
+    CONF_TEMP_INITIAL,
+    CONF_TEMP_MAX,
+    CONF_TEMP_MIN,
+    CONF_TEMP_STATE_TEMPLATE,
+    CONF_TEMP_STATE_TOPIC,
+    DEFAULT_OPTIMISTIC,
+)
+from .debug_info import log_messages
+from .mixins import MQTT_ENTITY_COMMON_SCHEMA, async_setup_entry_helper
+from .models import MqttCommandTemplate, MqttValueTemplate, ReceiveMessage
+from .util import get_mqtt_data, valid_publish_topic, valid_subscribe_topic
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "MQTT Water Heater"
+
+MQTT_WATER_HEATER_ATTRIBUTES_BLOCKED = frozenset(
+    {
+        water_heater.ATTR_CURRENT_TEMPERATURE,
+        water_heater.ATTR_MAX_TEMP,
+        water_heater.ATTR_MIN_TEMP,
+        water_heater.ATTR_TEMPERATURE,
+        water_heater.ATTR_OPERATION_LIST,
+        water_heater.ATTR_OPERATION_MODE,
+    }
+)
+
+VALUE_TEMPLATE_KEYS = (
+    CONF_CURRENT_TEMP_TEMPLATE,
+    CONF_MODE_STATE_TEMPLATE,
+    CONF_TEMP_STATE_TEMPLATE,
+)
+
+COMMAND_TEMPLATE_KEYS = {
+    CONF_MODE_COMMAND_TEMPLATE,
+    CONF_TEMP_COMMAND_TEMPLATE,
+}
+
+
+TOPIC_KEYS = (
+    CONF_CURRENT_TEMP_TOPIC,
+    CONF_MODE_COMMAND_TOPIC,
+    CONF_MODE_STATE_TOPIC,
+    CONF_TEMP_COMMAND_TOPIC,
+    CONF_TEMP_STATE_TOPIC,
+)
+
+
+_PLATFORM_SCHEMA_BASE = MQTT_BASE_SCHEMA.extend(
+    {
+        vol.Optional(CONF_CURRENT_TEMP_TEMPLATE): cv.template,
+        vol.Optional(CONF_CURRENT_TEMP_TOPIC): valid_subscribe_topic,
+        vol.Optional(CONF_MODE_COMMAND_TEMPLATE): cv.template,
+        vol.Optional(CONF_MODE_COMMAND_TOPIC): valid_publish_topic,
+        vol.Optional(
+            CONF_MODE_LIST,
+            default=[
+                STATE_ECO,
+                STATE_ELECTRIC,
+                STATE_GAS,
+                STATE_HEAT_PUMP,
+                STATE_HIGH_DEMAND,
+                STATE_PERFORMANCE,
+                STATE_OFF,
+            ],
+        ): cv.ensure_list,
+        vol.Optional(CONF_MODE_STATE_TEMPLATE): cv.template,
+        vol.Optional(CONF_MODE_STATE_TOPIC): valid_subscribe_topic,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_OPTIMISTIC, default=DEFAULT_OPTIMISTIC): cv.boolean,
+        vol.Optional(CONF_PAYLOAD_ON, default="ON"): cv.string,
+        vol.Optional(CONF_PAYLOAD_OFF, default="OFF"): cv.string,
+        vol.Optional(CONF_PRECISION): vol.In(
+            [PRECISION_TENTHS, PRECISION_HALVES, PRECISION_WHOLE]
+        ),
+        vol.Optional(CONF_RETAIN, default=DEFAULT_RETAIN): cv.boolean,
+        vol.Optional(CONF_TEMP_INITIAL, default=110): cv.positive_int,
+        vol.Optional(CONF_TEMP_MIN, default=DEFAULT_MIN_TEMP): vol.Coerce(float),
+        vol.Optional(CONF_TEMP_MAX, default=DEFAULT_MAX_TEMP): vol.Coerce(float),
```

Comment:
```suggestion
        vol.Optional(CONF_TEMP_MIN): vol.Coerce(float),
        vol.Optional(CONF_TEMP_MAX): vol.Coerce(float),
```

Reply:
It does seem a bit presumptuous to assign a default min/max temp. Seems like it should be required on a per device basis.

Reply:
Have a look at the top level class on how defaults are assigned. I believe this should work similar to the new climate code. I live in a country where deg C is the common unit. With the current code the heater will end beding targetted to 110 deg C, that is above the boiling point and not realistic. So the Fahrenheit default need to be converted if the values are not set by the user. So also the docs PR for mqtt climate.